pm/recipient.go

```diff
@@ -393,9 +396,15 @@ func (r *recipient) senderNoncesCleanupLoop() {
 			glog.Error(err)
 		case latestL1Block := <-sink:
 			r.senderNoncesLock.Lock()
-			for recipientRand, sn := range r.senderNonces {
-				if sn.expirationBlock.Cmp(latestL1Block) <= 0 {
-					delete(r.senderNonces, recipientRand)
+			for recipientRand, nonceMap := range r.senderNonces {
+				for _, key := range nonceMap.Keys() {
```

Comment:
If we track a single expiration block per recipientRandHash then we wouldn't need the nested loop here and instead could just clear all senderNonce values for the recipientRandHash once the expirationBlock is reached.